src/FakeItEasy/Core/FakeManagerProvider.cs

```diff
@@ -81,13 +81,19 @@ public void EnsureInitialized(object proxy)
                 {
                     this.initializedFakeManager = this.fakeManagerFactory(this.typeOfFake, proxy);
 
-                    this.fakeManagerAccessor.TagProxy(proxy, this.initializedFakeManager);
+                    this.fakeManagerAccessor.SetFakeManager(proxy, this.initializedFakeManager);
 
                     this.ApplyInitialConfiguration(proxy);
                 }
             }
         }
 
+        public void EnsureManagerIsRegistered()
+        {
+            var accessor = ServiceLocator.Current.Resolve<IFakeManagerAccessor>();

```

Comment:
OK, I agree that this test probably doesn't bring any value. Let's drop it, and serialize the `fakeManagerAccessor`.